Sources/CacheAdvance/CacheReader.swift

```diff
@@ -60,11 +60,17 @@ final class CacheReader {
             return message
 
         case .emptyRead:
+            guard !previousReadWasEmpty else {
+                // If the previous read was also empty, then the file has been corrupted.
+                // Two empty reads in a row means that offsetInFileAtEndOfNewestMessage is incorrect.
+                // This inconsistency likely is likely due to a crash occurring during a message write.
+                throw CacheAdvanceError.fileCorrupted
+            }
```

Comment:
To make sure I follow: we never expect to be in this state anymore, right? It could be useful to make clear that this is a safety net.

Reply:
yeah let me update this comment 👍